ddlproc/ddlprocessor.cpp

```diff
@@ -725,23 +414,393 @@ void DDLProcessor::process()
                         }
                     }
 
-                    dbrm.committed(txnid);
-                    txnid = dbrm.newTxnID(sessionID, true, true);
+                    fDbrm->committed(fTxnid);
+                    fTxnid = fDbrm->newTxnID(fSessionID, true, true);
 
-                    if (!txnid.valid)
+                    if (!fTxnid.valid)
                     {
                         throw std::runtime_error( std::string("Unable to start a transaction. Check critical log.") );
                     }
                 }
 
-                handler.fIos = ios;
-                handler.fByteStream = bs;
-                handler.fPackageType = packageType;
-                handler.fTxnid = txnid;
-                handler.fDbrm = &dbrm;
-                handler.fQtc = fQtc;
-                handler.fOamCache = oam::OamCache::makeOamCache();
-                fDdlPackagepool.invoke(handler);
+                processStatement();
+            }
+        }
+        catch (const std::exception &ex)
+        {
+            fIos.close();
```

Comment:
Shouldn't we re-throw so that the upper level catch was able to try to send a result. It might be dumb but at least we don't change the behavior of DMLProc towards external entities.